### ISSUE TITLE
Add a flag to specify resampling during template registration

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#3261]
### Context

In `sct_register_to_template`, the data are resampled by default to 1mm iso before registration:

https://github.com/neuropoly/spinalcordtoolbox/blob/900d29753117246fbb39d3ea07d642725a004b23/spinalcordtoolbox/scripts/sct_register_to_template.py#L441

This is done because the PAM50 template is 1mm iso. However, if users are using another template (which is possible with the flag `-t`), the resampling factor should be modifiable, especially use the destination template resolution is drastically different, as is the case for small animals. See notably this [user's request](https://forum.spinalcordmri.org/t/rat-cord-registration-to-template/641). 

### Suggestion

Implement a flag `-resample` to specify the resolution. To be minimally invasive and respect the syntax of `sct_resample`, I suggest using the following input syntax: `-resample 1x1x1` for 1mm iso. Or `-resample 0.1x0.1x0.1` for 0.1mm iso. So, "mm" would be the default unit.

**Comment by mdbudde** [comment]
Given these details (thanks), it seems to make more sense to simply obtain the resolution of the template being provided and use that properly formatted for the resample_file call.
It would avoid having to specify or have the user find/recall the resolution of the provided template and could be more robust to different templates.

**Comment by jcohenadad** [comment]
> Given these details (thanks), it seems to make more sense to simply obtain the resolution of the template being provided and use that properly formatted for the resample_file call.
> It would avoid having to specify or have the user find/recall the resolution of the provided template and could be more robust to different templates.

It is a good suggestion. So I assume the resolution of your rat template is 0.1mm iso?

**Comment by mdbudde** [comment]
It's not quite that, but close to it.  1mm resampling threw the error, and 0.1mm didn't. But I didn't check others besides those or see how it affected the final registration.  I see the PAM50 in the 5.2 release seems to be 0.5mm.  I wasn't sure how much the precision mattered for this step though and I'll defer to your experiences.

fslsize /usr/local/sct_v5.2/data/PAM50/template/PAM50_t2.nii.gz 
dim1		141
dim2		141
dim3		991
dim4		1
pixdim1		0.500000
pixdim2		0.500000
pixdim3		0.500000
pixdim4		1.000000

The home-built rat template I was using was:
fslsize ~/github/BuddeDoDMRIprocessing/Templates/RatHistoAtlas/template/RatHistoAtlas_t2.nii.gz 
dim1		151
dim2		128
dim3		580
dim4		1
pixdim1		0.050000
pixdim2		0.050000
pixdim3		0.150000
pixdim4		1.000000

**Comment by jcohenadad** [comment]
Right, I forgot the PAM50 was actually 0.5mm iso. The reason we resample to 1mm iso instead of 0.5mm is because of speed. It does take a _lot_ of time to straighten the spinal cord at 0.5mm iso. Also note that while the data are resampled to 1mm iso for the straightening, the registration to the template is done at 0.5mm iso, and there is no resampling error introduced by the intermediate resampling to 1mm iso because the warp_curve2straight transformation is concatenated to the subsequent non-linear transformations, starting here:
https://github.com/neuropoly/spinalcordtoolbox/blob/900d29753117246fbb39d3ea07d642725a004b23/spinalcordtoolbox/scripts/sct_register_to_template.py#L657

Sorry, I know this is a bit complicated. But the important message here is: the downsampling to 1mm iso is not really a problem in terms of precision when registering to a template that is 0.5mm iso.

**However**, what I wrote above concerns anatomical data acquired at ~1mm iso, which is usually the case for human anatomical data at 3T. If people acquire human data at very high res (eg 0.5mm iso), e.g. on a 7T scanner, they might want to benefit from this high resolution for an improved registration accuracy, in which case the resampling should be ~0.5mm iso (instead of 1mm). 

I should also add that this resampling is not performed when using the flag `-ref subject`. However, if you would like to straighten the cord before registration, then you should use `-ref template`. 

Now, coming back to the best strategy, I do think that allowing to specify the resampling via a flag might be the best approach. It would accommodate the above scenario of ppl acquiring high-res data for registration to the PAM50 template, and it would also accommodate scenario like yours, where you want to register to another template, but not necessarily stick to the resolution of the template for the straightening stage. E.g., in your case, 0.1x0.1x0.3mm might work as well (and could be much faster, if that's of any concern).
